src/kafka-util/src/client.rs

```diff
@@ -323,13 +343,75 @@ impl<C> TunnelingClientContext<C> {
     /// and restarts.
     pub async fn add_broker_rewrite(&self, broker: BrokerAddr, rewrite: BrokerRewrite) {
         let mut rewrites = self.rewrites.lock().await;
-        rewrites.insert(broker, Arc::new(move || rewrite.clone()));
+        rewrites.insert(
+            broker,
+            RewriteStatus::Rewritten(None, Arc::new(move || rewrite.clone())),
+        );
     }
 
     /// Returns a reference to the wrapped context.
     pub fn inner(&self) -> &C {
         &self.inner
     }
+
+    /// Returns a _consolidated_ `SshTunnelStatus` that communicates the status
+    /// of all active ssh tunnels `self` knows about.
+    pub async fn tunnel_statuses(&self) -> SshTunnelStatus {
+        // Pull out all the rewrites, temporarily.
+        let current_rewrites: Vec<_> = self
+            .rewrites
+            .lock()
+            .await
+            .iter()
+            .map(|(broker, status)| (broker.clone(), status.clone()))
+            .collect();
+
+        // First, get the statuses from broken _creation_ of ssh tunnels.
+        let mut ssh_statuses: BTreeMap<_, _> = current_rewrites
+            .iter()
+            .filter_map(|(broker, status)| match status {
+                RewriteStatus::CreationErrored(e) => Some((
+                    SshTunnelKey {
```

Comment:
This is an abstraction leak. `SshTunnelKey` is meant to be an internal implementation detail of the `SshTunnelManager`. Why do you need the key here at all? It seems like you could just have a list of statuses and that would be sufficient. Nothing actually looks at the keys, and there shouldn't be duplicates AFAICT because broker addresses must be unique (or, at least, it's extremely rare to have duplicate broker addresses, and if you do that, who cares if we check the status of an SSH tunnel twice).

Reply:
that sounds fine!